From 8c4e24783982fa200c1e72ed5e39e7c679a96bc9 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:42 -0400
Subject: [tools] perf/tests: Fix memory leak in dso-data.c

Message-id: <1408462094-14194-174-git-send-email-jolsa@redhat.com>
Patchwork-id: 88148
O-Subject: [PATCH RHEL7.1 BZ1131394 173/325] perf tests: Fix memory leak in dso-data.c
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 1df9297c8535a5bb2b776381e63d8334f87d4abe
Author: Felipe Pena <felipensp@gmail.com>
Date: Wed Oct 9 23:00:38 2013 -0300

description
===========
Fix for a memory leak on test_file() function in dso-data.c.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/tests/dso-data.c b/tools/perf/tests/dso-data.c
index dffe055..9cc81a3 100644
--- a/tools/perf/tests/dso-data.c
+++ b/tools/perf/tests/dso-data.c
@@ -35,6 +35,7 @@ static char *test_file(int size)
  if (size != write(fd, buf, size))
   templ = NULL;
 
+ free(buf);
  close(fd);
  return templ;
 }
-- 
1.7.1